src/json.ts

```diff
@@ -490,130 +509,193 @@ export function getJsonRpcIdValidator(options?: JsonRpcValidatorOptions) {
    * the value was serializable and a number of bytes that it will use when serialized to JSON.
    */
   function getJsonSerializableInfo(
-    value: unknown,
+    rawValue: unknown,
     skipSizing: boolean,
-  ): [isValid: boolean, plainTextSizeInBytes: number] {
+  ): ValidatedJson {
+    const value = getJsonValue(rawValue);
+
     if (value === undefined) {
-      return [false, 0];
+      return INVALID_JSON;
     } else if (value === null) {
       // Return already specified constant size for null (special object)
-      return [true, skipSizing ? 0 : JsonSize.Null];
+      return {
+        valid: true,
+        result: value,
+        size: skipSizing ? 0 : JsonSize.Null,
+      };
     }
 
     // Check and calculate sizes for basic (and some special) types
-    const typeOfValue = typeof value;
-    try {
-      if (typeOfValue === 'function') {
-        return [false, 0];
-      } else if (typeOfValue === 'string' || value instanceof String) {
-        return [
-          true,
-          skipSizing
-            ? 0
-            : calculateStringSize(value as string) + JsonSize.Quote * 2,
-        ];
-      } else if (typeOfValue === 'boolean' || value instanceof Boolean) {
-        if (skipSizing) {
-          return [true, 0];
-        }
+    if (typeof value === 'function') {
+      return INVALID_JSON;
+    } else if (typeof value === 'string' || value instanceof String) {
+      return {
+        valid: true,
+        // An instance of `String` is not assignable to our `Json` type.
+        result: String(value),
+        size: skipSizing
+          ? 0
+          : calculateStringSize(String(value)) + JsonSize.Quote * 2,
+      };
+    } else if (typeof value === 'boolean' || value instanceof Boolean) {
+      if (skipSizing) {
+        return {
+          valid: true,
+          // An instance of `Boolean` is not assignable to our `Json` type.
+          // eslint-disable-next-line eqeqeq
+          result: value == true,
+          size: 0,
+        };
+      }
+
+      return {
+        valid: true,
+        // An instance of `Boolean` is not assignable to our `Json` type.
         // eslint-disable-next-line eqeqeq
-        return [true, value == true ? JsonSize.True : JsonSize.False];
-      } else if (typeOfValue === 'number' || value instanceof Number) {
-        if (skipSizing) {
-          return [true, 0];
-        }
-        return [true, calculateNumberSize(value as number)];
-      } else if (value instanceof Date) {
-        if (skipSizing) {
-          return [true, 0];
-        }
-        return [
-          true,
-          // Note: Invalid dates will serialize to null
-          isNaN(value.getDate())
-            ? JsonSize.Null
-            : JsonSize.Date + JsonSize.Quote * 2,
-        ];
+        result: value == true,
+        // eslint-disable-next-line eqeqeq
+        size: value == true ? JsonSize.True : JsonSize.False,
```

Comment:
```suggestion
        size: skipSizing ? 0 : value == true ? JsonSize.True : JsonSize.False,
```

Then you can avoid the if-statement above, unless nested ternaries are not allowed.

Reply:
Parentheses make nested ternaries clearer:
```suggestion
        size: skipSizing ? 0 : (value == true ? JsonSize.True : JsonSize.False),
```

Reply:
ESLint doesn't allow nested ternaries:

> ESLint: Do not nest ternary expressions.(no-nested-ternary)